rpcserver.go

```diff
@@ -1685,7 +1685,7 @@ func handleGetBlockTemplateLongPoll(s *rpcServer, longPollID string, useCoinbase
 	// be manually unlocked before waiting for a notification about block
 	// template changes.
 
-	if err := state.updateBlockTemplate(s, useCoinbaseValue); err != nil {
+	if err := state.updateBlockTemplate(s, useCoinbaseValue, s.server.sigCache); err != nil {
```

Comment:
No change would be needed here if `updateBlockTemplate` accessed `s.server.sigCache` directly.